### PR TITLE
BUG: [GH 11738] Fix for end_time when multiple of a frequency is requested

### LINKED ISSUE #11738: BUG: end_time for Period with multiplied freq
end_time of a Period with multiplied freq is the square of the multiplication

```python
>>> pd.__version__
u'0.17.0'

>>> p = pd.Period('2015-08-01', freq='3D')
>>> p.end_time, p.to_timestamp(how='E')
(Timestamp('2015-08-09 23:59:59.999999999'), Timestamp('2015-08-03 00:00:00'))

>>> p = pd.Period('2015-08-01', freq='5D')
>>> p.end_time, p.to_timestamp(how='E')
(Timestamp('2015-08-25 23:59:59.999999999'), Timestamp('2015-08-05 00:00:00'))
```

### PR DESCRIPTION
**Pull request by srib**
closes #11738 
1. Added a new test with the multiple of a frequency. 
2. Rearranged some tests to improve readability of the code.

### CODE REVIEW

**Comment by jreback** [comment]
looks good. can you add a whatsnew note (bug fix), ping when green

File: pandas/tseries/tests/test_period.py [hunk 1]
```diff
@@ -566,19 +566,25 @@ def _ex(*args):
...
         xp = _ex(2012, 1, 3)
-        self.assertEqual(Period('2012', freq='B').end_time, xp)
+        self.assertEqual(xp, p.end_time)
 
+        p = Period('2012', freq='W')
         xp = _ex(2012, 1, 2)
-        self.assertEqual(Period('2012', freq='W').end_time, xp)
+        self.assertEqual(xp, p.end_time)
+
+        p = Period('2012', freq='15D')
```
Comment (jreback) [inline pandas/tseries/tests/test_period.py]: add the issue number as a comment here

**Comment by srib** [comment]
@jreback Sorry, didn't follow contributing guidelines carefully in this PR. Please see PR #11779.

**Comment by jreback** [comment]
ok, np. yeah, just keeping the original PR is much better / easier to follow the flow

**Pull request closed without merging**